CompilerSource/compiler/components/write_globals.cpp

```diff
@@ -62,70 +62,71 @@ int lang_CPP::compile_writeGlobals(EnigmaStruct* es, parsed_object* global)
-    wto << "  bool isSizeable = " << es->gameSettings.allowWindowResize << ";" << endl;
-    wto << "  bool showBorder = " << !es->gameSettings.dontDrawBorder << ";" << endl;
-    wto << "  bool showIcons = " << !es->gameSettings.dontShowButtons << ";" << endl;
-    wto << "  bool freezeOnLoseFocus = " << es->gameSettings.freezeOnLoseFocus << ";" << endl;
-    wto << "  bool treatCloseAsEscape = " << es->gameSettings.treatCloseAsEscape << ";" << endl;
-    wto << "  bool isFullScreen = " << es->gameSettings.startFullscreen << ";" << endl;
-    wto << "  string gameInfoText = \"" << esc(es->gameInfo.gameInfoStr) << "\";" << endl;
-    wto << "  string gameInfoCaption = \"" << es->gameInfo.formCaption << "\";" << endl;
-    wto << "  int gameInfoBackgroundColor = " << javaColor(es->gameInfo.backgroundColor) << ";" << endl;
-    wto << "  int gameInfoLeft = " << es->gameInfo.left << ";" << endl;
-    wto << "  int gameInfoTop = " << es->gameInfo.top << ";" << endl;
-    wto << "  int gameInfoWidth = " << es->gameInfo.width << ";" << endl;
-    wto << "  int gameInfoHeight = " << es->gameInfo.height << ";" << endl;
-    wto << "  bool gameInfoEmbedGameWindow = " << es->gameInfo.embedGameWindow << ";" << endl;
-    wto << "  bool gameInfoShowBorder = " << es->gameInfo.showBorder << ";" << endl;
-    wto << "  bool gameInfoAllowResize = " << es->gameInfo.allowResize << ";" << endl;
-    wto << "  bool gameInfoStayOnTop = " << es->gameInfo.stayOnTop << ";" << endl;
-    wto << "  bool gameInfoPauseGame = " << es->gameInfo.pauseGame << ";" << endl;
-    wto << "  int viewScale = " << es->gameSettings.scaling << ";" << endl;
-    wto << "  int windowColor = " << javaColor(es->gameSettings.colorOutsideRoom) << ";" << endl;
-    wto << "}" << endl;
-
-    for (parsed_object::globit i = global->globals.begin(); i != global->globals.end(); i++)
-      wto << i->second.type << " " << i->second.prefix << i->first << i->second.suffix << ";" << endl;
-    //This part needs written into a global object_parent class instance elsewhere.
-    //for (globit i = global->dots.begin(); i != global->globals.end(); i++)
-    //  wto << i->second->type << " " << i->second->prefixes << i->second->name << i->second->suffixes << ";" << endl;
-    wto << endl;
-
-    wto << "namespace enigma" << endl << "{" << endl << "  struct ENIGMA_global_structure: object_locals" << endl << "  {" << endl;
-    for (deciter i = dot_accessed_locals.begin(); i != dot_accessed_locals.end(); i++) // Dots are vars that are accessed as something.varname.
-      wto << "    " << i->second.type << " " << i->second.prefix << i->first << i->second.suffix << ";" << endl;
-
-    wto << "    ENIGMA_global_structure(const int _x, const int _y): object_locals(_x,_y) {}" << endl << "  };" << endl << "  object_basic *ENIGMA_global_instance = new ENIGMA_global_structure(global,global);" << endl << "}";
-    wto << endl;
+  wto << license;
```

Comment:
I looked over it and I don't think I did. I originally opened it to find out why the CLI was segfaulting at first, and it was just because game info didn't have a string allocated or something. It will be nice when all of this is rewritten with protos instead of having proto2es because then none of this code will crash, because protobuf returns valid defaults.

Reply:
Please review #1211 so I can get it merged and then merge branch master into this pr resolving your concerns.